drivers/bluetooth/hci/ipc.c

```diff
@@ -174,8 +175,15 @@ static struct net_buf *bt_ipc_iso_recv(const uint8_t *data, size_t remaining)
 		remaining -= sizeof(hdr);
 
 		net_buf_add_mem(buf, &hdr, sizeof(hdr));
+
+		fail_cnt = 0U;
 	} else {
-		LOG_ERR("No available ISO buffers!");
+		if ((fail_cnt % 100U) == 0U) {
```

Comment:
@Thalley, I think I understand the problem, but I am afraid that the solution looks like a debug hack. I have used myself this technique many times during debug sessions, but I think it could be possibly unpractical to use this in all places where same problem may occur in Zephyr code base.
Possibly the logging system should be updated, such that cannot exceed certain cpu usage limits, is it possible? Or introduce a new macro to reduce rate of potentially intensive logging messages, e.g. `LOG_ERR_REDUCED_RATE(rate, msg, …)`
Another proposal is to report the error only once, indicating this in the log message or using LOG_DBG as suggested by @jori-nordic in another thread.

Reply:
Will continue it here (and answer @jori-nordic). 

w.r.t using LOG_DBG, then you are left with the option of either no being aware of dropped packets, or CPU locking the system. Neither of those are really a good solution IMO. 

> I think I understand the problem, but I am afraid that the solution looks like a debug hack. I have used myself this technique many times during debug sessions, but I think it could be possibly unpractical to use this in all places where same problem may occur in Zephyr code base.

I agree that this is a debug hack. I would, however, argue that ISO logging is different than nearly any other logging, given the nature of ISO (continuous streams at low interval).  

> Possibly the logging system should be updated, such that cannot exceed certain cpu usage limits, is it possible? 

It should be. I don't fully understand why this can even happen today, as it should be possible to handle a log statement every e.g. 10ms in a low priority thread like the logging thread, but evidently it's not. 

> Another proposal is to report the error only once

That was my initial solution as well, basically replacing the count here with a `bool`. I did, however, feel that it was useful to know whether this occured once (not a big issue) or 100 times (big issue), and thus why I ended up with the counter. 

> Or introduce a new macro to reduce rate of potentially intensive logging messages, e.g. LOG_ERR_REDUCED_RATE(rate, msg, …)

That would be a general solution to this problem. I wouldn't be opposed to that if others feel that is would be a good way.

Reply:
> all places where same problem may occur in Zephyr code base

I agree in principle. But also we engineers are often guilty of inventing an whole abstraction for "the future".

I propose we merge this hack (as emil says, this is a special case for iso) and start designing the `REDUCED_RATE` macro the moment we have to duplicate this hack in the tree.

@Thalley Could you open a bug in the logging subsystem for the CPU usage?

Reply:
I am good with the proposal, and with reasoning behind it.